### PR TITLE
feat: added upgrade page

### LINKED ISSUE #13283: [CAL-2955] generic app.cal.com/upgrade page that explains what to upgrade
null

<sub>[CAL-2955](https://linear.app/calcom/issue/CAL-2955/generic-appcalcomupgrade-page-that-explains-what-to-upgrade)</sub>

### PR DESCRIPTION
**Pull request by PeerRich**
CleanShot 2024-01-17 at 20 28 06@2x

and if all good:

CleanShot 2024-01-17 at 20 28 28@2x

### CODE REVIEW

File: apps/web/pages/upgrade.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    onError: (error) => {
+      showToast(error.message, "error");
+    },
+  });
+
+  /* TODO: figure out how data works 
+  if (!data) return null;
+
+  const [membership] = data;
+  if (!membership) return null; */
```
Comment (PeerRich) [inline apps/web/pages/upgrade.tsx:36]: couldnt figure this out since the OrgUpgradeBanner got some generic {...props} with data, not sure what data is.

**Review by joeauyeung — DISMISSED** [review]
Button directs user to Stripe checkout page if they have an org to upgrade.

**Comment by keithwillcode** [comment]
@joeauyeung looks like we have type checks failing

**Review by zomars — APPROVED** [review]
fixed types in UsersTable. Was missing an icon import

**Pull request merged**